### PR TITLE
XR: changed snapping

### PR DESCRIPTION
**Pull request by VladStepura**
Fixed bugs with snapping and changed logic a little bit

### CODE REVIEW

File: src/js/xr/src/three/GPUPickers/GPUPickerHelper.js [hunk 1]
```diff
@@ -74,14 +87,15 @@ class GPUPickerHelper
...
-
+        
         return [{ object: this.pickedObject, point: canvasPos, distance: camera.worldPosition().distanceTo(canvasPos)}];
     }
     
     readRenderedPixel(renderer, scene, camera, renderTarget, pixelBuffer)
     {
         renderer.setRenderTarget(renderTarget);
         renderer.render(scene, camera);
+        renderer.render(scene, camera);
```
Comment (audionerd) [inline src/js/xr/src/three/GPUPickers/GPUPickerHelper.js]: This calls `.render` twice in a row? Is that intentional?

File: src/js/xr/src/use-interactions-manager.js [hunk 2]
```diff
@@ -773,7 +791,8 @@ const useInteractionsManager = ({
         onDragObjectEntry: (context, event) => {
           let controller = gl.vr.getController(context.draggingController)
           let object = event.intersection.object
-
+          controller.userData.draggedObject = object.uuid
```
Comment (audionerd) [inline src/js/xr/src/use-interactions-manager.js]: We already store the dragged object id as `context.selection`

File: src/js/xr/src/use-interactions-manager.js [hunk 3]
```diff
@@ -448,12 +457,17 @@ const useInteractionsManager = ({
     // include all interactables (Model Object, Character, etc)
     let list = scene.__interaction
 
+    // Checks if any object is dragged
```
Comment (audionerd) [inline src/js/xr/src/use-interactions-manager.js]: Could use `context.selection` here instead.

File: src/js/xr/src/use-interactions-manager.js [hunk 4]
```diff
@@ -855,7 +863,6 @@ const useInteractionsManager = ({
             object.userData.staticRotation = null
           }
 
-          controller.userData.selectOffset = null
```
Comment (audionerd) [inline src/js/xr/src/use-interactions-manager.js]: `selectOffset` is no longer cleared when we're done?

File: src/js/xr/src/use-interactions-manager.js [hunk 5]
```diff
@@ -105,19 +106,28 @@ const snapObjectRotation = object => {
...
   degreeY = THREE.Math.degToRad(Math.round(degreeY) * 22.5) * sign
-
   // snap rotation z to 180°
   let degreeZ = THREE.Math.radToDeg(Math.abs(object.rotation.z)) / 180
   degreeZ = THREE.Math.degToRad(Math.round(degreeZ) * 180)
 
+  let euler = new THREE.Euler(0, degreeY, 0)
+  euler.reorder('YXZ')
+
+/*   object.parent.worldToLocal(point)
```
Comment (audionerd) [inline src/js/xr/src/use-interactions-manager.js]: can we remove this comment and remove the `point` argument?

File: src/js/xr/src/use-interactions-manager.js (hunk 4 above)
Comment (VladStepura) [reply src/js/xr/src/use-interactions-manager.js]: We can keep but the thing is that objects don't use it anymore cause objects added to controller when they are selected. So there is no need for offset and if we apply it to object through our formula with the cursor we will get doubled rotation cause now object follows the rotation of the controller and cursor follows the rotation of the controller.

Comment (audionerd) [reply src/js/xr/src/use-interactions-manager.js]: Oh ok. So we could remove all references to selectOffset now?

Comment (VladStepura) [reply src/js/xr/src/use-interactions-manager.js]: Well, character still use it so I don't think we can completely remove it yet

**Pull request merged**